### DISCUSSION TITLE
Preventing downloading mp4 video files

### DISCUSSION THREAD

**Discussion by BrijeshDMIsystems** [ant-media/Ant-Media-Server#6083] [Q&A]
We seek guidance on implementing measures to prevent users from downloading mp4 video files. We are running ANT Media Server 2.8.2 on AWS for VOD play.

**Comment by Mohit-3196** [comment]
Hi @BrijeshDMIsystems,
How are you?

Let me write here the response that I sent to you on your support ticket yesterday.

In general, it's very difficult to prevent the download of MP4 files, and even if we use some type of mechanism, the MP4 files can still be downloaded. Even if using stream security to secure the MP4 file playback, once the user has the token it's downloadable. Though using a one-time token can help with the issue but still even that's not a foolproof solution as well.
As a workaround solution for this, I will recommend using HLS to record and play the VoD instead of MP4. It will also give you better control over the recordings as well as it'll be more efficient due to segmentation. Though using some browser extensions the HLS files files can be downloaded as well.

https://antmedia.io/docs/guides/playing-live-stream/hls-playing/#save-hls-records

One of our hackathon participants developed a DVR solution which I think you can use for this purpose. So, let me know if you would like to try that and I can arrange a meet.

Thank you.
